### PR TITLE
Merge `profile_limiters` stencils into `remap_profile`

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Purpose

This PR moves stencils from the `profile_limiters` file into `remap_profile`, and merges several of them to reduce the overall number of stencils.

Remove the sections below which do not apply.

## Code changes:

- Merge stencils from `profile_limiters.py` into `remap_profile.py`
- Remove `profile_limiters.py`

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Review by oelbert — APPROVED** [review]
Just one or two minor questions, but this looks really good! Thanks for the hard work

File: fv3core/stencils/remap_profile.py [hunk 1]
```diff
@@ -46,30 +41,123 @@ def constrain_interior(q, gam, a4):
     )
 
 
+@gtscript.function
+def ppm_constraint(
```
[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply fv3core/stencils/remap_profile.py:45]: Sounds good, changed.

File: fv3core/stencils/remap_profile.py [hunk 2]
```diff
@@ -163,433 +242,387 @@ def Apply_constraints(q: sd, gam: sd, a4_1: sd, a4_2: sd, a4_3: sd, iv: int):
...
+            a4_4 = 3.0 * x0
+            ext5 = abs(x0) > x1
+            ext6 = abs(a4_4) > x1
 
 
 @gtstencil()
-def set_top_as_iv0(a4_1: sd, a4_2: sd, a4_3: sd, a4_4: sd):
-    with computation(PARALLEL):
-        with interval(0, 1):
+def set_top(
```
[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply fv3core/stencils/remap_profile.py:268]: Next PR :-)? For now I thought top, inner, and bottom were reasonable boundaries.

File: fv3core/stencils/remap_profile.py [hunk 3]
```diff
@@ -163,433 +242,387 @@ def Apply_constraints(q: sd, gam: sd, a4_1: sd, a4_2: sd, a4_3: sd, iv: int):
...
-            a4_4 = 3 * (2 * a4_1 - (a4_2 + a4_3))
-
-
-@gtstencil()
-def set_top_as_iv2(a4_1: sd, a4_2: sd, a4_3: sd, a4_4: sd):
+            if iv == -1:
+                a4_4 = 3 * (2 * a4_1 - (a4_2 + a4_3))
+    # set_top_as_iv2
     with computation(PARALLEL):
         with interval(0, 1):
```
[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply fv3core/stencils/remap_profile.py:293]: Hmm, good question. I did not realize `if` statements were allowed outside `interval` blocks.

[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply fv3core/stencils/remap_profile.py:293]: Sadly, not yet supported,
```python
with computation(PARALLEL):
    if iv == -1:
        with interval(0, 1):
            a4_2 = 0.0 if a4_2 * a4_1 <= 0.0 else a4_2
        with interval(...):
            a4_4 = 3 * (2 * a4_1 - (a4_2 + a4_3))
```
gives error,
```
gt4py.frontend.gtscript_frontend.GTScriptSyntaxError: Invalid 'computation' specification at line...
```

**Comment by eddie-c-davis** [comment]
launch jenkins

File: fv3core/stencils/remap_profile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (oelbert) [reply fv3core/stencils/remap_profile.py:293]: Sorry, I should have been more clear. Right now it looks like there are two `with computation(PARALLEL), interval(0, 1):` sections near each other, and I was curious if it made sense to merge them to one section:
```
        with interval(0, 1):
            if iv == 0:
                a4_4 = 3 * (2 * a4_1 - (a4_2 + a4_3))
            elif iv == 2:
                ...
        with interval(...):
```

**Comment by rheacangeo** [comment]
launch jenkins

File: fv3core/stencils/remap_profile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply fv3core/stencils/remap_profile.py:293]: Oh I see what you mean, the _intervals_ should be mergeable. I shall try that.

**Pull request merged**